### ISSUE TITLE
Animated gif upload does not show animation

### ISSUE THREAD

**Issue by webtrend** [craftcms/aws-s3#31]
When I upload animated gif using webUI, the uploaded file does not show animation.

When I use a third party tool to upload the same file, animation works. 

Here are the examples of the same file, uploaded using Craft and other tool respectively. 
https://d1ybdyvhkcqsml.cloudfront.net/static_img/webimg/screnshot52.gif
https://d1ybdyvhkcqsml.cloudfront.net/static_img/webimg/screnshot54.gif

I am at loss of words on how this can happen. Craft plugin and the other tool I am using must be utilizing the same AWS API calls. How is it possible is beyond my understanding

**Comment by brandonkelly** [comment]
If installing ImageMagick isn’t an option, you can disable the [transformGifs](https://docs.craftcms.com/v3/config/config-settings.html#transformgifs) config setting, which will tell Craft to never make any changes to GIF files. Downside is that it’s less secure if someone uploads a GIF with an embedded malicious script, and also means that GIFs will always be the exact size that they were uploaded as, rather than resized based on transform properties.

**Comment by bzin** [comment]
I have both `ImageMagick` and `GD` installed in the server according to my PHP info. Is there any way of defining which one Craft uses? Because now when I upload a gif without a loop, the sanitization of the image removes it.

**Comment by brandonkelly** [comment]
Yes, using the [imageDriver](https://docs.craftcms.com/v3/config/config-settings.html#imagedriver) config setting.

```php
// -- config/general.php --
'imageDriver' => 'imagick',
```
